src/Reflection/Node.php

```diff
@@ -32,14 +35,16 @@ public function __construct($value, ?self $parent = null)
         }
     }
 
+    // phpcs:disable
     /**
      * Set parent node
      *
      * @param \Laminas\Server\Reflection\Node $node
      * @param  bool $new Whether or not the child node is newly created
      * and should always be attached
      */
-    public function setParent(Node $node, bool $new = false): void
+    // phpcs:enable
```

Comment:
Why are you having phpcs ignore this block?

Reply:
CS validation will fail because of the FQCN (expecting `Node`). The DocBlock param type is currently used in reflection. Change it to `Node` will fail test `ReflectionMethodTest::testMethodDocBlockFromParent`. With argument types in place, I would like to check whether it is possible to drop the DocBlock reflection in favor of using the argument types instead. I will have to do some research to see, if it will work (problematic could be mixed types).

Reply:
Ah, right - forgot that the component had not yet been updated to get type information via Reflection instead of the docblocks!

I think that will likely work really well. For mixed or union types, you still need annotations, so we can very likely fallback to parsing annotations only if the typehint is missing.

But agreed - that's for another set of changes.